src/uwtools/api/config.py

```diff
@@ -125,10 +125,12 @@ def realize(
     """
     NB: This docstring is dynamically replaced: See realize.__doc__ definition below.
     """
+    if update_config is None and update_format is not None:  # i.e. updates will be read from stdin
+        update_config = _ensure_data_source(update_config, stdin_ok)
```

Comment:
Previously, with the call to `_ensure_data_source()` inline on line 131, the `update_config` value was *always* being checked so that, if it was `None` and `stdin_ok` was `False` (the default), an exception would be raised. But it's completely valid for no update values to be provided because no update is desired, in which case the exception would be incorrect.

In order for `update_config` to be read from `stdin`, `update_format` must be provided, as `uwtools` would not otherwise know how to parse the stream. So, the presence of `update_format` implies that, if `update_config` isn't provided, the user wants to read it from `stdin`, and those criteria can be used to trigger the `stdin_ok` check for `update_config`.

Things are simpler for `input_config` because "no input" is not a valid configuration like "no update" is.